### DISCUSSION TITLE
PR checks won't run after action pushes changed code

### DISCUSSION THREAD

**Discussion by lukepighetti** [stefanzweifel/git-auto-commit-action#285] [Help] [answered]
I've noticed that if stefanzweifel/git-auto-commit-action pushes changed code, the PR checks won't run again to check it. Is there a way to do this?

**Comment by stefanzweifel** [comment, marked as answer]
Yeah, this happens due to a limitation set by GitHub for the `GITHUB_TOKEN`. From the [docs](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#using-the-github_token-in-a-workflow).

> When you use the repository's GITHUB_TOKEN to perform tasks on behalf of the GitHub Actions app, events triggered by the GITHUB_TOKEN will not create a new workflow run. This prevents you from accidentally creating recursive workflow runs.

You can fix this by creating a new [Personal Access Token (PAT)](https://github.com/settings/tokens/new),
storing the token as a secret in your repository and then passing the new token to the [`actions/checkout`](https://github.com/actions/checkout#usage) Action step.

```yaml
- uses: actions/checkout@v3
  with:
    token: ${{ secrets.PAT }}
```

If you create a personal access token, apply the `repo` and `workflow` scopes.
If you work in an organization and don't want to create a PAT from your personal account, we recommend using a [robot account](https://docs.github.com/en/github/getting-started-with-github/types-of-github-accounts) for the token.

Commits made by git-auto-commit will then trigger other workflows again (eg. running your test suite). Keep in mind that this constellation could lead to and endless chain of workflow executions.
